src/robot/variables/replacer.py

```diff
@@ -139,31 +139,31 @@ def _get_variable_item(self, match, value):
         for item in match.items:
             if is_dict_like(value):
                 value = self._get_dict_variable_item(name, value, item)
-            elif is_list_like(value):
-                value = self._get_list_variable_item(name, value, item)
+            elif is_iterable(value):
+                value = self._get_iterable_variable_item(name, value, item)
             else:
                 raise VariableError(
-                    "Variable '%s' is %s, not list or dictionary, and thus "
+                    "Variable '%s' is %s, not a dictionary or iterable, and thus "
```

Comment:
I'm slightly worried is "dictionary or iterable" (or the old "list or dictionary") clear for an average user. Should we consider rephrasing this somehow? One alternatively could be something like "Variable '%s' does not support key or index access", but I'm not sure is that much better. Opinions? Better ideas?

Another thing to consider is enhancing the error to tell how this problem can be avoided by escaping the `[item]` part. Should we add something like "To use '[%s]' as a literal value, it needs to be escaped like '\\[%s]'.", where `%s` are placeholders for `item`, to the end of the error message?

I know changing error messages mean extra work updating existing tests, but we've got lot of questions and bug reports since we added the `${var}[item]` syntax in RF 3.1 and I think making the situation more clear to users is a good thing.

Reply:
I am changing "not a dictionary or iterable" to "which is not subscriptable" and enhancing the error message as you suggested. Would this extension to the error message also be useful in case of a subscriptable variable that is not a sequence? I am not familiar with the use cases of `\[%s]`.

Reply:
Shouldn't `\[%s]` be added to the [user guide](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#escaping)?